chat-plugins/dexsearch.js

```diff
@@ -11,13 +11,149 @@
 
 const RESULTS_MAX_LENGTH = 10;
 
-exports.commands = {
-	ds: 'dexsearch',
-	dsearch: 'dexsearch',
-	dexsearch: function (target, room, user, connection, cmd, message) {
-		if (!this.canBroadcast()) return;
+if (!process.send) {
+	// This is the parent
+	Tools.dexsearchProcess = require('child_process').fork('dexsearch.js', {cwd: __dirname});
+
+	let resolvers = {};
+	let queryId = 1;
+	let queryChild = function (query) {
+		let localQueryId = queryId++;
+		query.id = localQueryId;
+		return new Promise((resolve, reject) => {
+			resolvers[localQueryId] = resolve;
+			Tools.dexsearchProcess.send(query);
+		});
+	};
+	Tools.dexsearchProcess.on("message", message => resolvers[message.id](message.value));
+
+	exports.commands = {
+		ds: 'dexsearch',
+		dsearch: 'dexsearch',
+		dexsearch: function (target, room, user, connection, cmd, message) {
+			if (!this.canBroadcast()) return;
+
+			if (!target) return this.parse('/help dexsearch');
+
+			let msg = (this.broadcasting ? "" : message);
+
+			let promise = queryChild({target: target, cmd: 'dexsearch', canAll: (!this.broadcasting || room.isPersonal), message: msg});
```

Comment:
This object should be multiline.

You can also just do `queryChild(...).then(...)` instead of needing a promise variable.